diff --git a/dist/core.mjs b/dist/core.mjs
index f1b72a7d771fed564c4669d3a26051f6c19f010a..cc3ff2914eb4361a9e180d7edf4b76db175ff48d 100644
--- a/dist/core.mjs
+++ b/dist/core.mjs
@@ -989,6 +989,7 @@ function createTransformerFactory(defaultTwoslasher, defaultRenderer) {
     const trigger = explicitTrigger instanceof RegExp ? explicitTrigger : /\btwoslash\b/;
     if (!renderer)
       throw new ShikiTwoslashError("Missing renderer");
+    const map = /* @__PURE__ */ new WeakMap();
     const filter = options.filter || ((lang, _, options2) => langs.includes(lang) && (!explicitTrigger || trigger.test(options2.meta?.__raw || "")));
     return {
       preprocess(code) {
@@ -997,27 +998,30 @@ function createTransformerFactory(defaultTwoslasher, defaultRenderer) {
           lang = langAlias[this.options.lang];
         if (filter(lang, code, this.options)) {
           const twoslash = twoslasher(code, lang, twoslashOptions);
-          this.meta.twoslash = twoslash;
+          map.set(this.meta, twoslash);
           this.options.lang = twoslash.meta?.extension || lang;
           return twoslash.code;
         }
       },
       tokens(tokens) {
-        if (!this.meta.twoslash)
+        const twoslash = map.get(this.meta);
+        if (!twoslash)
           return;
         return splitTokens(
           tokens,
-          this.meta.twoslash.nodes.flatMap(
+          twoslash.nodes.flatMap(
             (i) => ["hover", "error", "query", "highlight", "completion"].includes(i.type) ? [i.start, i.start + i.length] : []
           )
         );
       },
       pre(pre) {
-        if (this.meta.twoslash)
-          this.addClassToHast(pre, "twoslash lsp");
+        const twoslash = map.get(this.meta);
+        if (!twoslash)
+          return;
+        this.addClassToHast(pre, "twoslash lsp");
       },
       code(codeEl) {
-        const twoslash = this.meta.twoslash;
+        const twoslash = map.get(this.meta);
         if (!twoslash)
           return;
         const insertAfterLine = (line, nodes) => {
@@ -1068,6 +1072,8 @@ function createTransformerFactory(defaultTwoslasher, defaultRenderer) {
             continue;
           }
           const tokens = locateTextTokens(node.line, node.character, node.length);
+          if (!tokens.length)
+            throw new ShikiTwoslashError(`Cannot find tokens for node: ${JSON.stringify(node)}`);
           const wrapTokens = (fn) => {
             const line = this.lines[node.line];
             let charIndex = 0;
diff --git a/package.json b/package.json
index add5a8aa82c81bd836398f48e338f0993dde95cb..da642a7c89f316d0253529bbe4dd15d3ca45e92b 100644
--- a/package.json
+++ b/package.json
@@ -48,9 +48,15 @@
     "*.css",
     "dist"
   ],
+  "scripts": {
+    "build": "unbuild",
+    "dev": "unbuild --stub",
+    "prepublishOnly": "nr build",
+    "test": "vitest"
+  },
   "dependencies": {
-    "twoslash": "^0.1.2",
-    "@shikijs/core": "1.0.0"
+    "@shikijs/core": "1.0.0",
+    "twoslash": "^0.1.2"
   },
   "devDependencies": {
     "@iconify-json/carbon": "^1.1.28",
@@ -58,10 +64,5 @@
     "@shikijs/twoslash": "^3.1.2",
     "hast-util-from-html": "^2.0.1",
     "typescript": "^5.3.3"
-  },
-  "scripts": {
-    "build": "unbuild",
-    "dev": "unbuild --stub",
-    "test": "vitest"
   }
 }
