diff --git a/dist/client.mjs b/dist/client.mjs
index be74201c8cd1419d5a556aa15feb1d949ec50245..c6dd263287bdbed501ff393bc93a0014741bc280 100644
--- a/dist/client.mjs
+++ b/dist/client.mjs
@@ -12,7 +12,8 @@ const TwoSlashFloatingVue = {
           delay: 0,
           handleResize: false,
           autoHide: true,
-          instantMove: true
+          instantMove: true,
+          flip: false
         },
         ...options.theme
       }
diff --git a/dist/index.d.mts b/dist/index.d.mts
index b8d2b28208de11de68e91f02186be695a3a72df0..60f48bd968cc3c75bb38e2e6f433147e2d8b294a 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -1,9 +1,9 @@
-import { RendererRichOptions, TwoSlashRenderers, TransformerTwoSlashOptions } from 'shikiji-twoslash';
+import { RendererRichOptions, TwoSlashRenderer, TransformerTwoSlashOptions } from 'shikiji-twoslash';
 export { defaultHoverInfoProcessor } from 'shikiji-twoslash';
 import { ShikijiTransformer } from 'shikiji';
 import { VitePressPluginTwoSlashOptions as VitePressPluginTwoSlashOptions$1 } from 'vitepress-plugin-twoslash';
 
-declare function rendererFloatingVue(options?: VitePressPluginTwoSlashOptions$1 & RendererRichOptions): TwoSlashRenderers;
+declare function rendererFloatingVue(options?: VitePressPluginTwoSlashOptions$1 & RendererRichOptions): TwoSlashRenderer;
 
 interface VitePressPluginTwoSlashOptions extends TransformerTwoSlashOptions, RendererRichOptions {
     /**
diff --git a/dist/index.d.ts b/dist/index.d.ts
index b8d2b28208de11de68e91f02186be695a3a72df0..60f48bd968cc3c75bb38e2e6f433147e2d8b294a 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,9 +1,9 @@
-import { RendererRichOptions, TwoSlashRenderers, TransformerTwoSlashOptions } from 'shikiji-twoslash';
+import { RendererRichOptions, TwoSlashRenderer, TransformerTwoSlashOptions } from 'shikiji-twoslash';
 export { defaultHoverInfoProcessor } from 'shikiji-twoslash';
 import { ShikijiTransformer } from 'shikiji';
 import { VitePressPluginTwoSlashOptions as VitePressPluginTwoSlashOptions$1 } from 'vitepress-plugin-twoslash';
 
-declare function rendererFloatingVue(options?: VitePressPluginTwoSlashOptions$1 & RendererRichOptions): TwoSlashRenderers;
+declare function rendererFloatingVue(options?: VitePressPluginTwoSlashOptions$1 & RendererRichOptions): TwoSlashRenderer;
 
 interface VitePressPluginTwoSlashOptions extends TransformerTwoSlashOptions, RendererRichOptions {
     /**
diff --git a/dist/index.mjs b/dist/index.mjs
index 9202ccb428abf8dddb4f7cd3636681486bccdb03..4e4c02061bc8de13d5d38ac5c13d5832a08d9a61 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1,4 +1,6 @@
-import { rendererRich, defaultHoverInfoProcessor, transformerTwoSlash } from 'shikiji-twoslash';
+import { createTransformerFactory } from 'shikiji-twoslash/core';
+import { createTwoSlasherVue } from 'twoslash-vue';
+import { rendererRich, defaultHoverInfoProcessor } from 'shikiji-twoslash';
 export { defaultHoverInfoProcessor } from 'shikiji-twoslash';
 import { gfmFromMarkdown } from 'mdast-util-gfm';
 import { fromMarkdown } from 'mdast-util-from-markdown';
@@ -23,7 +25,8 @@ function rendererFloatingVue(options = {}) {
       {
         ...this.options,
         transformers: [],
-        transforms: void 0
+        transforms: void 0,
+        lang: this.options.lang === "vue" ? "tsx" : this.options.lang
       }
     ).children[0].children;
     themedContent.forEach((node2) => {
@@ -121,7 +124,10 @@ function rendererFloatingVue(options = {}) {
 }
 
 function transformerTwoslash(options = {}) {
-  const twoslash = transformerTwoSlash({
+  const twoslash = createTransformerFactory(
+    createTwoSlasherVue()
+  )({
+    langs: ["ts", "tsx", "js", "jsx", "json", "vue"],
     explicitTrigger: true,
     renderer: rendererFloatingVue(),
     ...options
@@ -139,6 +145,10 @@ function transformerTwoslash(options = {}) {
           options2.transformers?.splice(options2.transformers.indexOf(vPre), 1);
       }
       return twoslash.preprocess.call(this, code, options2);
+    },
+    postprocess(html, options2) {
+      if (this.meta.twoslash && options2.lang === "vue")
+        return html.replace(/{/g, "&#123;");
     }
   };
 }
diff --git a/package.json b/package.json
index 10be3386e04940d41cd4f3645f30e48b9e1b2580..090eb997bc951188c1d6599dd88b23bd3ef9e9ad 100644
--- a/package.json
+++ b/package.json
@@ -53,6 +53,7 @@
     "mdast-util-gfm": "^3.0.0",
     "mdast-util-to-hast": "^13.1.0",
     "vue": "^3.4.11",
+    "twoslash-vue": "^0.0.1",
     "shikiji-twoslash": "0.9.19",
     "shikiji": "0.9.19"
   },
diff --git a/style-core.css b/style-core.css
index 03008e8ec54fe54aea7c4a4e188ff68d9bfa9b62..d9d6332a92be21e8a49e09278c173db62227147e 100644
--- a/style-core.css
+++ b/style-core.css
@@ -4,6 +4,10 @@
   --twoslash-border-color: #8884;
 }
 
+.v-popper--theme-twoslash {
+  z-index: calc(var(--vp-z-index-local-nav) - 1);
+}
+
 .v-popper--theme-twoslash .v-popper__inner {
   background: var(--vp-c-bg);
   color: var(--vp-c-text-1);
diff --git a/style.css b/style.css
index 589000a2b6c7dfc6573ea70a5e526a577011f21f..dd3053cfbc5dafb033a06bc40c97207431c4a516 100644
--- a/style.css
+++ b/style.css
@@ -210,6 +210,10 @@
   --twoslash-border-color: #8884;
 }
 
+.v-popper--theme-twoslash {
+  z-index: calc(var(--vp-z-index-local-nav) - 1);
+}
+
 .v-popper--theme-twoslash .v-popper__inner {
   background: var(--vp-c-bg);
   color: var(--vp-c-text-1);